packages/rpc-methods/src/restricted/manageAccounts.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  SubjectType,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+  PermissionSpecificationBuilder,
+} from '@metamask/permission-controller';
+import { Json, JsonStruct, NonEmptyArray } from '@metamask/utils';
+import {
+  assert,
+  string,
+  object,
+  optional,
+  union,
+  array,
+  record,
+  Infer,
+} from 'superstruct';
+
+const SnapMessageStruct = object({
+  method: string(),
+  params: optional(union([array(JsonStruct), record(string(), JsonStruct)])),
+});
+
+type Message = Infer<typeof SnapMessageStruct>;
+
+export const methodName = 'snap_manageAccounts';
+
+export type ManageAccountsMethodHooks = {
+  /**
+   * Gets the snap keyring implementation.
+   */
+  getSnapKeyring: (snapOrigin: string) => Promise<{
+    handleKeyringSnapMessage: (
+      snapId: string,
+      message: Message,
+    ) => Promise<Json>;
+  }>;
+
+  /**
+   * Saves the snap keyring, should be called after mutable operations.
+   *
+   * If an account was removed the address of the account must be passed to
```

Comment:
This comment is not up to date

Reply:
Done in [5375fcb](https://github.com/MetaMask/snaps/pull/1290/commits/5375fcb116a942f7cdb54936e5a47816a6fcee18)